src/controller_main.c

```diff
@@ -105,8 +105,42 @@
 #include "pb_decode.h"
 #include "polygon.h"
 #include "rfc7539.h"
+#include "ui_events.h"
 #include "ui_instruction.h"
 
+ui_event_t os_ui_event;
```

Comment:
I believe this is only required for testing flows now, so we should probably add a comment here so that we know that we need to remove this function after revamp is complete